integrations/dynatrace/main.py

```diff
@@ -0,0 +1,61 @@
+from enum import StrEnum
+from typing import Any
+
+from loguru import logger
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+from client import DynatraceClient
+
+
+class ObjectKind(StrEnum):
+    PROBLEM = "problem"
+    SLO = "slo"
+    ENTITY = "entity"
+
+
+def initialize_client() -> DynatraceClient:
+    return DynatraceClient(
+        ocean.integration_config["dynatrace_host_url"],
+        ocean.integration_config["dynatrace_api_key"],
+    )
+
+
+@ocean.on_resync(ObjectKind.PROBLEM)
+async def on_resync_problems(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    dynatrace_client = initialize_client()
+
+    async for problems in dynatrace_client.get_problems():
+        logger.info(f"Received batch with {len(problems)} problems")
+        yield problems
+
+
+@ocean.on_resync(ObjectKind.SLO)
+async def on_resync_slos(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    dynatrace_client = initialize_client()
+
+    async for slos in dynatrace_client.get_slos():
+        logger.info(f"Received batch with {len(slos)} SLOs")
+        yield slos
+
+
+@ocean.on_resync(ObjectKind.ENTITY)
+async def on_resync_entities(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    dynatrace_client = initialize_client()
+
+    async for entities in dynatrace_client.get_entities():
+        logger.info(f"Received batch with {len(entities)} entities")
+        yield entities
+
+
+@ocean.router.post("/webhook")
+async def on_problem_event(event: dict[str, str | Any]) -> dict[str, bool]:
+    dynatrace_client = initialize_client()
+    logger.info(f"Received problem event: {event}")
+
+    if problem_id := event.get("ProblemID"):
+        problem = await dynatrace_client.get_single_problem(problem_id)
+        await ocean.register_raw(ObjectKind.PROBLEM, [problem])
+
+    logger.info("Webhook event processed")
+    return {"ok": True}
```

Comment:
you are missing `on_startup` webhook initialization. 
its ok if its not possible as it requires specific configuration. we will need to make sure we have clear definition on how to create the webhook.
Either way I would suggest making the url more type specific to `/webhook/event` as there might be more than event webhook, if this is not the case, feel free to correct me.